### PR TITLE
Check if relayer funds are enough to process submitted transaction (estimates).

### PR DESCRIPTION
**Pull request by piohei**
Transaction cost is estimated based on gas limit and current gas cost estimates based on chain.

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -251,6 +252,31 @@ impl RelayerApi {
             ));
         }
 
+        let relayer_queued_tx_gas_limit_sum = app
+            .db
+            .get_relayer_pending_txs_gas_limit_sum(api_token.relayer_id())
+            .await?;
+
+        let block_fees = app.db.get_latest_block_fees_by_chain_id(relayer.chain_id).await?;
+        if let Some(block_fees) = block_fees {
```
Comment (piohei) [inline src/server.rs:261]: I decided to check funds only if estimates are available. Other option is to return error here if we think it is crucial for us.

File: src/service.rs [hunk 2]
```diff
@@ -82,6 +82,18 @@ impl Service {
 
         Ok(())
     }
+
+    pub async fn is_estimates_ready_for_chain(&self, chain_id: u64) -> bool {
```
Comment (Dzejkop) [inline src/service.rs:86]:
```suggestion
    pub async fn are_estimates_ready_for_chain(&self, chain_id: u64) -> bool {
```

?

File: tests/common/service_builder.rs [hunk 3]
```diff
@@ -94,6 +94,22 @@ impl ServiceBuilder {
         let client =
             TxSitterClient::new(format!("http://{}", service.local_addr()));
 
+        // Awaits for estimates to be ready
+        let mut is_estimates_ready = false;
```
Comment (Dzejkop) [inline tests/common/service_builder.rs:98]:
```suggestion
        let mut are_estimates_ready = false;
```

File: src/service.rs [hunk 4]
```diff
@@ -82,6 +82,18 @@ impl Service {
...
+        let res = self
+            ._app
```
Comment (Dzejkop) [inline src/service.rs:88]: Please change struct field to `app`

**Review by Dzejkop — APPROVED** [review]

**Pull request merged**